include/pybind11/pybind11.h

```diff
@@ -472,7 +477,9 @@ class cpp_function : public function {
                 arg.value.dec_ref();
             }
             if (rec->def) {
-                std::free(const_cast<char *>(rec->def->ml_doc));
+                if (rec->def->ml_doc) {
```

Comment:
In principle, no braces necessary (cfr. other parts of this code); fine to keep them, though.

Reply:
Actually, `std::free` can be called on a `nullptr`: https://en.cppreference.com/w/cpp/memory/c/free says

> If ptr is a null pointer, the function does nothing.

So if anything, I'd put a comment that it could be `nullptr`, but I don't think that's necessary.

Reply:
Extra: potentially, you could reset to `nullptr` after freeing, to easy debugging when things go wrong. But it wasn't there, so I don't think this PR per se _needs_ to do that.